Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -811,6 +811,8 @@ void HTMLInputElement::attributeChanged(const QualifiedName& name, const AtomStr
             setFormControlValueMatchesRenderer(false);
         }
         updateValidity();
+        if (selfOrPrecedingNodesAffectDirAuto())
+            updateEffectiveDirectionalityOfDirAuto();
```

Comment:
Do we ever end up doing duplicate work with https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/Source/WebCore/html/HTMLInputElement.cpp#1180 ?

Reply:
I think this is fine actually, HTMLInputElement::setValue shouldn't trigger HTMLInputElement::attributeChanged